js/jquery.socialfeed.js

```diff
@@ -112,69 +178,20 @@ if (typeof Object.create !== 'function') {
             Feed[social_network].posts.push(this);
         }
         SocialFeedPost.prototype = {
-            render: function() {
-                var rendered_html = Feed.template(this.content);
-                var data = this.content;
-
-                if ($(container).children('[social-feed-id=' + data.id + ']').length !== 0) {
-                    return false;
-                }
-                if ($(container).children().length === 0) {
-                    $(container).append(rendered_html);
-                } else {
-                    var i = 0,
-                        insert_index = -1;
-                    $.each($(container).children(), function() {
-                        if ($(this).attr('dt-create') < data.dt_create) {
-                            insert_index = i;
-                            return false;
-                        }
-                        i++;
-                    });
-                    $(container).append(rendered_html);
-                    if (insert_index >= 0) {
-                        insert_index++;
-                        var before = $(container).children('div:nth-child(' + insert_index + ')'),
-                            current = $(container).children('div:last-child');
-                        $(current).insertBefore(before);
-                    }
-
-                }
-                if (options.media_min_width) {
-
-                    var query = '[social-feed-id=' + data.id + '] img.attachment';
-                    var image = $(query);
-
-                    // preload the image
-                    var height, width = '';
-                    var img = new Image();
-                    var imgSrc = image.attr("src");
-
-                    $(img).load(function() {
+            pushData: function() {
+                render_array.push(this.content);
 
-                        if (img.width < options.media_min_width) {
-                            image.hide();
-                        }
-                        // garbage collect img
-                        delete img;
-
-                    }).error(function() {
-                        // image couldnt be loaded
-                        image.hide();
+                loaded_post_count++;
+                if (loaded_post_count == posts_to_load_count) {
```

Comment:
this is aproblem i found in case one feed not reach the posts_to_load_count. For example a RSS feed that contain only 4 posts and the limit is put to 10, the render will not start. This problem happens also to the official version, i've opened a bug/request time ago. The problem will be fixed if there is a way to check if there are not other records froma a feed and even if is less then the posts_to_load_count but there are no more render it anyway